packages/webdriverio/src/commands/element/scrollIntoView.ts

```diff
@@ -43,74 +44,59 @@ function scrollIntoViewWeb (
  */
 export async function scrollIntoView (
     this: WebdriverIO.Element,
-    options: ScrollIntoViewOptions | boolean = { block: 'start', inline: 'nearest' }
+    options: ScrollIntoViewOptions & {scrollMode?: 'if-needed' | 'always'} | boolean = { block: 'start', inline: 'nearest', scrollMode: 'if-needed' }
 ) {
     const browser = getBrowserObject(this)
 
     /**
      * Appium does not support the "wheel" action
      */
     if (browser.isMobile) {
-        return scrollIntoViewWeb.call(this, options)
+        return scrollIntoViewWeb.call(this, options as any)
     }
+    const element = {
+        [ELEMENT_KEY]: this.elementId, // w3c compatible
+        ELEMENT: this.elementId, // jsonwp compatible
+    } as any as HTMLElement
+    let parsedOptions
 
+    if (typeof options === 'boolean') {
+        parsedOptions = options ? { scrollMode: 'always', block: 'start', inline: 'nearest' } : { scrollMode: 'always', block: 'end', inline: 'nearest' }
+    } else if (!options.scrollMode) {
+        parsedOptions = { ...options, ...{ scrollMode: 'if-needed' } }
+    } else {
+        parsedOptions = options
+    }
     try {
-        /**
-         * by default the WebDriver action scrolls the element just into the
-         * viewport. In order to stay complaint with `Element.scrollIntoView()`
-         * we need to adjust the values a bit.
-         */
-        const elemRect = await browser.getElementRect(this.elementId)
-        const viewport = await browser.getWindowSize()
-        let [scrollX, scrollY] = await browser.execute(() => [
-            window.scrollX, window.scrollY
-        ])
-
-        // handle elements outside of the viewport
-        scrollX = elemRect.x <= viewport.width ? elemRect.x : viewport.width / 2
-        scrollY = elemRect.y <= viewport.height ? elemRect.y : viewport.height / 2
-
-        const deltaByOption = {
-            start: { y: elemRect.y - elemRect.height, x: elemRect.x - elemRect.width },
-            center: { y: elemRect.y - Math.round((viewport.height - elemRect.height) / 2), x: elemRect.x - Math.round((viewport.width - elemRect.width) / 2) },
-            end: { y: elemRect.y - (viewport.height - elemRect.height), x: elemRect.x - (viewport.width - elemRect.width) }
+        const deltaPosition = (await browser.execute(
+            compute,
+            element,
+            parsedOptions as Record<string, any>
+        )) as {
+            el?: Element;
+            top: number;
+            left: number;
+            isVisible?: boolean
         }
-
-        let [deltaX, deltaY] = [deltaByOption.start.x, deltaByOption.start.y]
-        if (options === true) {
-            options = { block: 'start', inline: 'nearest' }
+        // eslint-disable-next-line unicorn/prefer-ternary
+        if (deltaPosition && 'isVisible' in deltaPosition && deltaPosition.isVisible && parsedOptions.scrollMode !== 'if-needed') {
```

Comment:
Can we add a comment why we do a execute scroll into view here?

Reply:
done